src/components/com_tjdashboard/administrator/tables/widgets.php

```diff
@@ -55,4 +55,45 @@ public function check()
 
 		return parent::check();
 	}
+
+	/**
+	 * Method to delete a Widget.
+	 *
+	 * @param   mixed  $pk  Primary key value to delete. Optional
+	 * 
+	 * @return  boolean  True on success.
+	 *
+	 * @since   1.0.0
+	 */
+	public function delete($pk = null)
+	{
+		JSession::checkToken() or jexit(JText::_('JINVALID_TOKEN'));
+		$result = 0;
+
+		if (!empty($pk))
+		{
+			try
+			{
+				$db = JFactory::getDbo();
+
+				// Check if widget is not default
+				$table = JTable::getInstance('widgets', 'TjdashboardTable', array('dbo', $db));
+
+				if ($table->load(array('dashboard_widget_id' => $pk)) && ($table->core != 1))
+				{
+					$result = parent::delete($pk);
+				}
+				elseif ($table->load(array('dashboard_widget_id' => $pk)) && ($table->core == 1))
+				{
+					JFactory::getApplication()->enqueueMessage(JText::_('COM_TJDASHBOARD_DEFAULT_WIDGETS_DELETE_ERROR_MESSAGE'), 'error');
```

Comment:
@shitalkakade24 Set the error. Don't enqueue message here

Reply:
done